central/cve/datastoretest/datastore_sac_test.go

```diff
@@ -588,19 +611,114 @@ func (s *cveDataStoreSACTestSuite) TestSACImageCVEGetBatch() {
 }
 
 func (s *cveDataStoreSACTestSuite) TestSACImageCVECount() {
-	s.T().Skip("Not implemented yet.")
+	err := s.dackboxTestStore.PushImageToVulnerabilitiesGraph(waitForIndexing)
+	defer s.cleanImageToVulnerabilitiesGraph(waitForIndexing)
+	s.Require().NoError(err)
+	for _, c := range imageCVETestCases {
+		s.Run(c.contextKey, func() {
+			if features.PostgresDatastore.Enabled() {
+				s.T().Skip("Skipping image count tests on postgres for now")
+			}
+			testCtx := s.imageTestContexts[c.contextKey]
+			count, err := s.imageCVEStore.Count(testCtx, nil)
+			s.NoError(err)
+			expectedCount := 0
+			for _, visible := range c.expectedCVEFound {
+				if visible {
+					expectedCount++
+				}
+			}
+			s.Equal(expectedCount, count)
+		})
+	}
 }
 
 func (s *cveDataStoreSACTestSuite) TestSACImageCVESearch() {
-	s.T().Skip("Not implemented yet.")
+	err := s.dackboxTestStore.PushImageToVulnerabilitiesGraph(waitForIndexing)
+	defer s.cleanImageToVulnerabilitiesGraph(waitForIndexing)
+	s.Require().NoError(err)
+	for _, c := range imageCVETestCases {
+		s.Run(c.contextKey, func() {
+
+			testCtx := s.imageTestContexts[c.contextKey]
+			results, err := s.imageCVEStore.Search(testCtx, nil)
+			s.NoError(err)
+			expectedCVENames := make([]string, 0, len(c.expectedCVEFound))
+			for name, visible := range c.expectedCVEFound {
+				if visible {
+					expectedCVENames = append(expectedCVENames, getImageCVEID(name))
+				}
+			}
+			fetchedCVEIDs := make(map[string]bool, 0)
```

Comment:
If it's only about storing the key value, the idiomatic thing to use within go would be `map[string]struct{}`. But looking at the code, a simple string array would suffice as well, right?

Applies to more places than just the one here.